mempool/cat/tx.go

```diff
@@ -34,3 +34,5 @@ func newWrappedTx(tx types.Tx, key types.TxKey, height, gasWanted, priority int6
 
 // Size reports the size of the raw transaction in bytes.
 func (w *wrappedTx) size() int64 { return int64(len(w.tx)) }
+
+func (w *wrappedTx) isReserved() bool { return w.height == -1 }
```

Comment:
[question] what is a reserved transaction? How does it relate to height == -1? Any links to docs would be helpful. 

Also this function could use a Go doc.

Reply:
reserved tx is a cat thing, I'm not sure yet if we need it as afaik it's an optimization

https://github.com/celestiaorg/celestia-core/blob/5cfce0b9e32f4d878bcb0e71c3367e401aafcbc2/mempool/cat/pool.go#L299

Reply:
It's basically a way of not processing the same transaction in parallel. Say I receive transaction x and while processing it I receive tx x again I will verify it again because I haven't yet added it to the mempool

Reply:
Rephrasing that: isReserved is true when a transaction has been received but hasn't been added to the mempool yet because it's still being processed? Does that mean isReserved == isProcessing?

Are these the exhaustive states a transaction could be in: https://github.com/celestiaorg/celestia-core/pull/1267/files#r1526067986

If yes, would it make sense to consolidate on one term either: isProcessing or isReserved?